packages/@aws-cdk/aws-appsync/lib/schema.ts

```diff
@@ -92,6 +100,52 @@ export class Schema {
     this.definition = `${this.definition}${sep}${addition}\n`;
   }
 
+  /**
+   * Add a query field to the schema's Query. If one isn't set by
+   * the user, CDK will create an Object Type called 'Query'. For example,
+   *
+   * type Query {
+   *   fieldName: Field.returnType
+   * }
+   *
+   * @param fieldName the name of the query
+   * @param field the resolvable field to for this query
+   */
+  public addQuery(fieldName: string, field: ResolvableField): ObjectType {
+    if (this.mode !== SchemaMode.CODE) {
+      throw new Error('API cannot add to schema\'s query because schema definition mode is not configured as CODE.');
```

Comment:
can we make this more descriptive:

something like 
```ts
`Unable to add query. Schema definition mode must be ${SchemaMode.CODE} Received: ${this.mode}`
```

reword as you like, but it's useful to share what was expected along with what was received.